plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/ui/navigator/model/AppEngineProjectElement.java

```diff
@@ -109,12 +145,20 @@ static boolean hasLayoutChanged(Collection<IFile> changedFiles) {
    */
   private IFile descriptorFile;
 
-  private AppEngineDescriptor descriptor;
+  private String projectId;
+  private String projectVersion;
+  private String serviceId;
+
+  /** Return the App Engine environment type (e.g., standard, flexible). */
```

Comment:
Looking at the code, this would be `flex` for the flexible and `null` for the standard, wouldn't it?